### PR TITLE
#239 PsLinkedin passes the request parameters to the URI.

### PR DESCRIPTION
**Pull request by super132**
For task #239, removing request parameters in `Href` object and move them into body.

### CODE REVIEW

**Comment by super132** [comment]
@davvd , this is fix for #239, please review. Thanks.

**Comment by davvd** [comment]
@super132 Thanks, let me find someone who can review this pull request

**Comment by davvd** [comment]
@caarlos0 please review, thanks

**Comment by caarlos0** [comment]
@super132 can you write some test?

**Comment by super132** [comment]
@caarlos0, I can write some tests but now there is no test for PsLinkedin and there is puzzle for it. Does it mean that I need to solve it together with that puzzle? Thanks.

**Comment by caarlos0** [comment]
@super132 oh, if there is a puzzle already, I think we are good to go here..

**Comment by caarlos0** [comment]
@rultor good to merge

**Comment by yegor256** [comment]
@super132 @caarlos0 no, we're not good to go. we should wait for that puzzle resolution and then proceed here with a unit test

**Comment by super132** [comment]
@yegor256 @caarlos0 understood. Let's wait for #158 to be resolved in order to add a test. Thanks.

**Comment by caarlos0** [comment]
@yegor256 @super132 makes sense.

**Comment by caarlos0** [comment]
@davvd we are waiting for #158 here.

**Comment by davvd** [comment]
> @davvd we are waiting for #158 here.

@caarlos0 sure, let's wait for #158

**Comment by davvd** [comment]
@super132 #158 is closed, it was an impediment

**Comment by super132** [comment]
@caarlos0 I have amended the login test case to check for request body and URL. Please have a look. Thanks.

**Comment by caarlos0** [comment]
@super132 looks good.

**Comment by caarlos0** [comment]
@rultor good to merge

File: src/test/java/org/takes/facets/auth/social/PsLinkedinTest.java [hunk 1]
```diff
@@ -67,9 +70,36 @@ public void logins() throws IOException {
         final Take take = new TkFork(
             new FkRegex(
                 "/uas/oauth2/accessToken",
+                // @checkstyle AnonInnerLengthCheck (100 lines)
                 new Take() {
                     @Override
                     public Response act(final Request req) throws IOException {
+                        final BufferedInputStream buffer =
```
Comment (yegor256) [inline src/test/java/org/takes/facets/auth/social/PsLinkedinTest.java]: @super132 I would rather use `RqPrint(req).printBody()`

**Comment by yegor256** [comment]
@super132 there is a better/shorter way to do what you're doing, see above

**Comment by super132** [comment]
@yegor256 thanks for you advice. I have amended it using `RqPrint`, please take a look. Thanks.

File: src/test/java/org/takes/facets/auth/social/PsLinkedinTest.java [hunk 2]
```diff
@@ -67,9 +70,32 @@ public void logins() throws IOException {
...
+                        MatcherAssert.assertThat(
+                            new RqPrint(req).printBody(),
+                            Matchers.stringContainsInOrder(
+                                Arrays.asList(
+                                    "grant_type=authorization_code",
+                                    new StringBuffer("client_id=")
+                                        .append(lapp)
+                                        .toString(),
+                                    "redirect_uri=",
+                                    new StringBuffer("client_secret=")
```
Comment (yegor256) [inline src/test/java/org/takes/facets/auth/social/PsLinkedinTest.java]: @super132 don't you think that `String.format()` is more suitable here? see http://www.yegor256.com/2014/06/19/avoid-string-concatenation.html

**Comment by yegor256** [comment]
@super132 one more comment, see above

**Comment by super132** [comment]
@yegor256 right, it would be cleaner to use `String.format()` instead. I have revised the code. Thanks.

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by davvd** [comment]
@elenavolokhova please, let us know what do you think about this ticket, according to our [QA rules](http://www.teamed.io/qa.html)

**Comment by davvd** [comment]
@rultor deploy now pls

**Comment by elenavolokhova** [comment]
@caarlos0
According to our [quality rules](http://www.teamed.io/qa.html):
 > The code reviewer found at least three problems in the code.
 
Please confirm that you will try to find at least three problems next time.

**Comment by caarlos0** [comment]
@elenavolokhova Yes, I'll try

**Comment by elenavolokhova** [comment]
@davvd Quality is acceptable here.

**Comment by davvd** [comment]
> @davvd Quality is acceptable here.

@elenavolokhova thanks, got it. everybody, please try to make it good next time

**Comment by davvd** [comment]
@caarlos0 just added 10 mins to @elenavolokhova (for QA), payment ID is `57255929`; paid, thanks, added **23 mins** to your account, payment `AP-1JP12351VU9246113`, 21 hours and 55 mins was spent; extra minutes for review comments (c=8); added +23 to your rating, now it is equal to [+1952](http://www.netbout.com/b/35095?open=rating)

**Pull request merged**